lib/dry/inflector.rb

```diff
@@ -34,20 +34,46 @@ def initialize(&blk)
       @inflections = Inflections.build(&blk)
     end
 
-    # Camelize a string
+    # Lower camelize a string
     #
     # @param input [String,Symbol] the input
-    # @return [String] the camelized string
+    # @return [String] the lower camelized string
     #
-    # @since 0.1.0
+    # @since 0.1.2
     #
     # @example
     #   require "dry/inflector"
     #
     #   inflector = Dry::Inflector.new
-    #   inflector.camelize("dry/inflector") # => "Dry::Inflector"
-    def camelize(input)
-      input.to_s.gsub(/\/(.?)/) { "::#{Regexp.last_match(1).upcase}" }.gsub(/(?:\A|_)(.)/) { Regexp.last_match(1).upcase }
+    #   inflector.camelize_lower("data_mapper") # => "dataMapper"
+    def camelize_lower(input)
+      internal_camelize(input, /(?:_)(.)/)
+    end
+
+    # Upper camelize a string
+    #
+    # @param input [String,Symbol] the input
+    # @return [String] the upper camelized string
+    #
+    # @since 0.1.2
+    #
+    # @example
+    #   require "dry/inflector"
+    #
+    #   inflector = Dry::Inflector.new
+    #   inflector.camelize_upper("data_mapper") # => "DataMapper"
+    def camelize_upper(input)
+      internal_camelize(input, /(?:\A|_)(.)/)
+    end
+
+    # @since 0.1.2
+    # @api private
```

Comment:
Why is this `@api private`?

Reply:
Because I think we should favor `camelize_upper` and `camelize_lower`. Leaving this just for back compatibility. What do you think?

Reply:
I think we should alias `camelize` with `camelize_upper`, as it's the "default" camelization.